hubblestack/extmods/hubble_mods/ssl_certificate.py

```diff
@@ -0,0 +1,315 @@
+        host_ip = endpoint_chained.get('host_ip')
+        host_port = endpoint_chained.get('host_port')
+    else:
+        host_ip = runner_utils.get_param_for_module(block_id, block_dict, 'host_ip')
+        host_port = runner_utils.get_param_for_module(block_id, block_dict, 'host_port')
+
+    ssl_timeout = runner_utils.get_param_for_module(block_id, block_dict, 'ssl_timeout', 3)
+    path = runner_utils.get_param_for_module(block_id, block_dict, 'path')
+
+    cert = _get_cert(host_ip, host_port, ssl_timeout=ssl_timeout) if host_ip else _get_cert(path, from_file=True)
+    if not cert:
+        return runner_utils.prepare_negative_result_for_module(block_id, 'unable_to_load_certificate')
+
+    log.info("FDG ssl_certificate - cert found, parsing certificate")
+    cert_details = _parse_cert(cert, host_ip, host_port, path)
+    stop_time = time.time()
+    cert_details['execution_time'] = stop_time - start_time
+    return runner_utils.prepare_positive_result_for_module(block_id, cert_details)
+
+
+def get_filtered_params_to_log(block_id, block_dict, extra_args=None):
+    """
+    For getting params to log, in non-verbose logging
+
+    :param block_id:
+        id of the block
+    :param block_dict:
+        parameter for this module
+    :param extra_args:
+        Extra argument dictionary, (If any)
+        Example: {'chaining_args': {'result': {'host_ip': '127.0.0.1',
+                                                'host_port': 443},
+                                    'status': True},
+                  'caller': 'Audit'}
+    """
+    log.debug('get_filtered_params_to_log for id: {0}'.format(block_id))
+    chain_args = extra_args.get('chaining_args') if extra_args else None
+    endpoint_chained = runner_utils.get_chained_param(chain_args)
+    if endpoint_chained:
+        host_ip = endpoint_chained.get('host_ip')
+        host_port = endpoint_chained.get('host_port')
+    else:
+        host_ip = runner_utils.get_param_for_module(block_id, block_dict, 'host_ip')
+        host_port = runner_utils.get_param_for_module(block_id, block_dict, 'host_port')
+
+    path = runner_utils.get_param_for_module(block_id, block_dict, 'path')
+    if path:
+        return {'path': path}
+    return {'host_ip': host_ip,
+            'host_port': host_port}
+
+
+def _get_cert(source, port=443, ssl_timeout=3, from_file=False):
+    cert = _get_cert_from_file(source) if from_file else _get_cert_from_endpoint(source, port, ssl_timeout)
+    return cert
+
+
+def _get_cert_from_endpoint(server, port=443, ssl_timeout=3):
+    try:
+        log.debug("ssl_certificate is checking for ssl cert on {0}:{1}".format(server, port))
+        hostport = (str(server), int(port))
+        setdefaulttimeout(ssl_timeout)
+        cert_details = ssl.get_server_certificate(hostport)
+    except Exception as e:
+        log.error('Unable to retrieve certificate from {0}. Error: {1}'.format(server, e))
+        cert_details = None
+    if not cert_details:
```

Comment:
We can remove this 'if' block. Return None from 'except' block itself.

Reply:
done

Reply:
done